### PR TITLE
Redesign Excel connector and verify on SLBeta1

### PR DESCRIPTION
**Pull request by Chuhaa**
## Purpose

- Migrate existing connector to slbeta1.
- Add new operations.
- Add display annotation.
- Add samples.
- Update documents.
- Update CI/CD pipelines.

## Goals
Fixes https://github.com/wso2-enterprise/choreo/issues/4735
Fixes https://github.com/wso2-enterprise/choreo/issues/6214

## Documentation
Please find the design document here https://docs.google.com/document/d/1JWBPeqmihNT-0R8HPbAevrBkWFdVACNw40xNRTC-P0M/edit?usp=sharing

## Automation tests
 - Unit tests 
    Tested
 - Integration tests
    Tested

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
```
import ballerina/log;
import ballerinax/microsoft.excel;

configurable string clientId = ?;
configurable string clientSecret = ?;
configurable string refreshToken = ?;
configurable string refreshUrl = ?;
configurable string workbookIdOrPath = ?;

excel:ExcelConfiguration configuration = {
    authConfig: {
        clientId: clientId,
        clientSecret: clientSecret,
        refreshToken: refreshToken,
        refreshUrl: refreshUrl
    }
};

excel:Client excelClient = check new (configuration);

public function main() {
    excel:Worksheet|error response = excelClient->addWorksheet(workbookIdOrPath, "sheetName");
    if (response is excel:Worksheet) {
        log:printInfo(response.toString());
    }
}
```

## Test environment
SLBeta 1
JDK 11
Ubuntu 20.04

### CODE REVIEW

File: excel/util.bal [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            }
+            path += element;
+        }
+    }
+    return path;
+}
+
+isolated function addQueryParameters(string path, Query query) returns string {
+    string queryPath = path + QUESTION_MARK;
+    if (query?.count is boolean) {
```
Comment (SanduDS) [inline excel/util.bal:46]: Can't we use Elvis operator here

Comment (Chuhaa) [reply excel/util.bal:46]: Yes. But here it does not have else part.  So is it good to use as below format here?
```
queryPath = query?.count is boolean ? queryPath + DOLLAR_SIGN + COUNT + EQUAL_SIGN + query?.count.toString() :  queryPath;
```

File: excel/util.bal [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        queryPath = queryPath + DOLLAR_SIGN + COUNT + EQUAL_SIGN + query?.count.toString();
+    }
+    if (query?.expand is string) {
+        queryPath = queryPath + DOLLAR_SIGN + EXPAND + EQUAL_SIGN + query?.expand.toString();
+    }
+    if (query?.filter is string) {
+        queryPath = queryPath + DOLLAR_SIGN + FILTER + EQUAL_SIGN + query?.filter.toString();
+    }
+    if (query?.format is string) {
+        queryPath = queryPath + DOLLAR_SIGN + FORMAT + EQUAL_SIGN + query?.count.toString();
```
Comment (sachinira) [inline excel/util.bal:56]: Seems like a typo here. Shouldn't it be `query?.format.toString()`  
https://docs.microsoft.com/en-us/graph/query-parameters#format-parameter

Comment (Chuhaa) [reply excel/util.bal]: Removed this method. Thanks

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Ballerina Build
+        uses: ballerina-platform/ballerina-action/@nightly
+        with:
+          args:
+            build -c --skip-tests
```
Comment (LakshanSS) [inline .github/workflows/pull-request.yml:16]: No need to skip tests

Comment (Chuhaa) [reply .github/workflows/pull-request.yml:16]: Ballerina platform secrets cannot be accessed by pull request job. So we can skip the tests here.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ *
+ * This is a general purpose Gradle build.
+ * Learn how to create Gradle builds at https://guides.gradle.org/creating-new-gradle-builds
+ */
```
Comment (LakshanSS) [inline build.gradle:6]: Nightly build github package publishing logic is missing

File: excel/Ballerina.toml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[package]
+org= "ballerinax"
+name= "microsoft.excel"
+version= "0.11.0"
```
Comment (LakshanSS) [inline excel/Ballerina.toml:4]: In master branch, version should be in SNAPSHOT

Comment (Chuhaa) [reply excel/Ballerina.toml]: Fixed in https://github.com/ballerina-platform/module-ballerinax-microsoft.excel/commit/7221c821c3249e73220eb70ef191a826e7684036

File: build.gradle (hunk 4 above)
Comment (Chuhaa) [reply build.gradle:6]: Fixed in https://github.com/ballerina-platform/module-ballerinax-microsoft.excel/commit/7221c821c3249e73220eb70ef191a826e7684036. Thanks

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+            url = 'https://repo.maven.apache.org/maven2'
+        }
+
+        maven {
+            url = 'https://maven.pkg.github.com/ballerina-platform/ballerina-lang'
+            credentials {
+                username System.getenv("packageUser")
+                password System.getenv("packagePAT")
+            }
+        }
```
Comment (LakshanSS) [inline build.gradle:40]: No need of this since this connector doesn't have any java code

Comment (Chuhaa) [reply build.gradle]: Fixed in https://github.com/ballerina-platform/module-ballerinax-microsoft.excel/commit/40d7de3b26e9a1c87a814add53b6621decd3f176. Thanks

**Review by abeykoon — APPROVED** [review]

**Pull request merged**